### DISCUSSION TITLE
Feedback on a proposal for helping tools find environments created by PDM

### DISCUSSION THREAD

**Discussion by brettcannon** [pdm-project/pdm#1110] [General]
I was hoping someone/@frostming could have a look at https://github.com/brettcannon/python-launcher/discussions/168 and let me know if there's anything in that proposal that wouldn't work for PDM? Specifically, is the proposal missing anything if PDM (or anyone for that matter) were to provide a `_py_launcher_pdm` entry point that allowed tools like the Python Launcher, VS Code, etc. to discover virtual environments that PDM created? I'm not explicitly asking anyone implement this idea right now, I'm just trying to do my homework before starting to do any coding around it.

**Comment by frostming** [comment]
PDM doesn't create environments itself but the plugin `pdm-venv` does, using whatever backend(virtualenv, venv,  conda) users choose.

PDM also has its own python discovery logic using [findpython](https://github.com/frostming/findpython). I have no objection implementing such an entry point and let's see how that can be improved.

**Reply by brettcannon** [reply]
Thanks for the info! I'll have a look at findpython and keep you in the loop as stuff progresses.
